File: GoldmanSachs/JavaScript/UglyNumber.js
// I'm starting to like JS more and more

var isUgly = function(n) {
    return ([...Array(6).keys()].reduce((final, key) => {
        if (key > 1 && final > 1) {
            while (final % key == 0) {
                final /= key;
            }
        }

        return final;
    }, n) == 1);
};

File: GoldmanSachs/CPP/UglyNumber.cpp
class Solution {
public:
    bool isUgly(int n) {
        if (n == 1) return true;
        if (n == 0 || n == -1) return false;

        for (int i = 2; i < 6; i++) {
            while (n % i == 0) {
                n /= i;
            }
        }

        return n == 1;
    }
};
